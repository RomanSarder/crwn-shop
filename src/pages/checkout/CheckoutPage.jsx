import React from 'react'
import { useSelector } from 'react-redux'
import styled from 'styled-components'
import CheckoutForm from '../../components/checkout-form/CheckoutForm'
import CheckoutItem from '../../components/checkout-item/CheckoutItem'
import TotalPrice from '../../components/styled/TotalPrice'
import { selectCartItems, selectCartItemsTotalPrice } from '../../store/cart/selectors'

var StyledCheckoutPage = styled.div`
    width: 95%;
    min-height: 90vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 5rem auto 0;

    @media (min-width: 992px) {
        width: 75%;
    }

    @media (min-width: 1200px) {
        width: 55%;
    }

    .total {
        margin-left: auto;
    }

    .header {
        width: 100%;
        height: 4rem;
        display: flex;
        justify-content: space-between;
        border-bottom: 1px solid darkgrey;
    }
`

var StyledHeaderBlock = styled.div`
    text-transform: capitalize;
    width: 23%;
    text-align: center;

    &:last-child {
        width: 8%
    }
`

export default function CheckoutPage() {
    var cartItems = useSelector(selectCartItems)
    var totalPrice = useSelector(selectCartItemsTotalPrice)

    return (
        <StyledCheckoutPage>
            <div className="header">
                <StyledHeaderBlock>
                    <span>Product</span>
                </StyledHeaderBlock>
                <StyledHeaderBlock>
                    <span>Description</span>
                </StyledHeaderBlock>
                <StyledHeaderBlock>
                    <span>Quantity</span>
                </StyledHeaderBlock>
                <StyledHeaderBlock>
                    <span>Price</span>
                </StyledHeaderBlock>
                <StyledHeaderBlock>
                    <span>Remove</span>
                </StyledHeaderBlock>
            </div>

            {cartItems.map(function renderCheckoutItems (item) {
                return <CheckoutItem item={item} key={item.id} />
            })}

            <TotalPrice className="total">Total: ${totalPrice}</TotalPrice>
            <CheckoutForm />
        </StyledCheckoutPage>
    )
}
