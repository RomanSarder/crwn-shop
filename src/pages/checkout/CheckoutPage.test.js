import { render, screen } from '@testing-library/react'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import CheckoutPage from './CheckoutPage'

var testCartItems = [
    {
        id: 2,
        name: 'test item 1',
        quantity: 1,
        price: 10,
        imageUrl: 'test image url'
    },
    {
        id: 3,
        name: 'test item 2',
        quantity: 3,
        price: 30,
        imageUrl: 'test image url'
    },
    {
        id: 4,
        name: 'test item 3',
        quantity: 1,
        price: 10,
        imageUrl: 'test image url'
    }
]

it('<CheckoutPage /> should render checkout items and total price', () => {
    var store = configureStore()({
        cart: {
            cartItems: testCartItems
        }
    })
    
    var { getByText, getAllByAltText, getAllByText, container } = render(
        <Provider store={store}>
            <CheckoutPage />
        </Provider>
    )
    expect(container).toMatchSnapshot()
    expect(getAllByText('test item', { exact: false }).length).toEqual(3)
    expect(getAllByAltText('test item', { exact: false }).length).toEqual(3)
    expect(getByText('Total: $110'))
})