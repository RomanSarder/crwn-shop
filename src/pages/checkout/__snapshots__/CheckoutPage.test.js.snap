// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutPage /> should render checkout items and total price 1`] = `
.c7 {
  min-width: 16.5rem;
  width: auto;
  height: 5rem;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 5rem;
  padding: 0 3.5rem 0 3.5rem;
  font-size: 1.5rem;
  background-color: black;
  color: white;
  text-transform: uppercase;
  font-family: 'Open Sans Condensed';
  font-weight: bolder;
  cursor: pointer;
  border: none;
}

.c7:hover {
  background-color: white;
  color: black;
  border: 1px solid black;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 40rem;
  gap: 1rem;
}

.c6 .card-field {
  padding: 1rem;
}

.c6 .test-warning {
  color: red;
  font-size: 3.2rem;
  text-align: center;
  margin: 2rem 0;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: none;
  border: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:focus {
  outline: none;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 10rem;
  border-bottom: 1px solid darkgrey;
  padding: 1.5rem 0;
  font-size: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .image-container {
  width: 23%;
  padding-right: 1.5rem;
}

.c2 .image-container img {
  width: 100%;
  max-height: 100%;
}

.c2 .name,
.c2 .quantity,
.c2 .price {
  width: 23%;
  text-align: center;
}

.c2 .quantity {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .quantity .c3 {
  font-size: 1.6rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 .remove-button {
  padding-left: 1.2rem;
  cursor: pointer;
}

.c2 .c3 {
  font-size: 2.6rem;
}

.c5 {
  text-transform: uppercase;
  font-weight: 700;
  font-size: 2.4rem;
}

.c0 {
  width: 95%;
  min-height: 90vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 5rem auto 0;
}

.c0 .total {
  margin-left: auto;
}

.c0 .header {
  width: 100%;
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid darkgrey;
}

.c1 {
  text-transform: capitalize;
  width: 23%;
  text-align: center;
}

.c1:last-child {
  width: 8%;
}

@media (min-width:992px) {
  .c0 {
    width: 75%;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 55%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="header"
    >
      <div
        class="c1"
      >
        <span>
          Product
        </span>
      </div>
      <div
        class="c1"
      >
        <span>
          Description
        </span>
      </div>
      <div
        class="c1"
      >
        <span>
          Quantity
        </span>
      </div>
      <div
        class="c1"
      >
        <span>
          Price
        </span>
      </div>
      <div
        class="c1"
      >
        <span>
          Remove
        </span>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="image-container"
      >
        <img
          alt="test item 1"
          src="test image url"
        />
      </div>
      <span
        class="name"
      >
        test item 1
      </span>
      <div
        class="quantity"
      >
        <button
          class="c3 c4"
        >
          <span>
            ❮
          </span>
        </button>
        <span
          class="value"
        >
          1
        </span>
        <button
          class="c3 c4"
        >
          <span>
            ❯
          </span>
        </button>
      </div>
      <span
        class="price"
      >
        10
      </span>
      <div
        class="remove-button"
      >
        <button
          class="c3 c4"
        >
          ✕
        </button>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="image-container"
      >
        <img
          alt="test item 2"
          src="test image url"
        />
      </div>
      <span
        class="name"
      >
        test item 2
      </span>
      <div
        class="quantity"
      >
        <button
          class="c3 c4"
        >
          <span>
            ❮
          </span>
        </button>
        <span
          class="value"
        >
          3
        </span>
        <button
          class="c3 c4"
        >
          <span>
            ❯
          </span>
        </button>
      </div>
      <span
        class="price"
      >
        30
      </span>
      <div
        class="remove-button"
      >
        <button
          class="c3 c4"
        >
          ✕
        </button>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="image-container"
      >
        <img
          alt="test item 3"
          src="test image url"
        />
      </div>
      <span
        class="name"
      >
        test item 3
      </span>
      <div
        class="quantity"
      >
        <button
          class="c3 c4"
        >
          <span>
            ❮
          </span>
        </button>
        <span
          class="value"
        >
          1
        </span>
        <button
          class="c3 c4"
        >
          <span>
            ❯
          </span>
        </button>
      </div>
      <span
        class="price"
      >
        10
      </span>
      <div
        class="remove-button"
      >
        <button
          class="c3 c4"
        >
          ✕
        </button>
      </div>
    </div>
    <span
      class="c5 total"
    >
      Total: $
      110
    </span>
    <form
      class="c6"
    >
      <span
        class="test-warning"
      >
        *Please use the following credit card for test payment*
        <br />
        4242 4242 4242 4242 - Exp: any future date
      </span>
      <div
        class="card-field"
      >
        <div />
      </div>
      <button
        class="c7"
        type="submit"
      >
        Check Out Now
      </button>
    </form>
  </div>
</div>
`;
